scripts/Engineering/texture/TextureUtils.py

```diff
@@ -398,13 +444,30 @@ def fit_all_peaks(wss: Sequence[str], peaks: Sequence[float], peak_window: float
             bg_func_name = "LinearBackground"
 
             func_generator = TexturePeakFunctionGenerator([])  # don't fix any parameters
+            parameters_to_tie = () if not parameters_to_tie else parameters_to_tie
             initial_function, md_fit_kwargs, intensity_estimates = func_generator.get_initial_fit_function_and_kwargs_from_specs(
-                ws, peak, (xmin, xmax), ("A", "B"), peak_func_name, bg_func_name
+                ws, peak, (xmin, xmax), parameters_to_tie, peak_func_name, bg_func_name
             )
 
-            Fit(
+            fit_object = Fit(
                 Function=initial_function,
                 CostFunction="Least squares",
+                Output="first_fit",
+                MaxIterations=10,  # if it hasn't fit in 10 it is likely because the texture has the peak missing
+                **md_fit_kwargs,
+            )
+
+            fit_result = {"Function": fit_object.Function.function, "OutputWorkspace": fit_object.OutputWorkspace.name()}
+
+            # update peak mask based on I/sig from fit
+            *_, i_over_sigma, _ = calc_intens_and_sigma_arrays(fit_result, "Hessian")
+            fit_mask = i_over_sigma > i_over_sigma_thresh
+
+            # fit only peak pixels and let peak centers vary independently of DIFC ratio
+            final_function = func_generator.get_final_fit_function(fit_result["Function"], fit_mask, 0.02)
+            Fit(
+                Function=final_function,
+                CostFunction="Least squares",
```

Comment:
Mantid crashes when fitting - I was wondering whether this is because at some point here's an empty bin in the data?
I think if you're using a weighted least squares cost-function you need to set `IgnoreInvalidData=True`? It doesn't look like this would be in `md_fit_kwargs`. This is good to do anyway, but it shouldn't crash Mantid...